segmentstore/server/src/main/java/io/pravega/segmentstore/server/tables/TableBucketReader.java

```diff
@@ -90,15 +89,14 @@
      * will exclude all {@link ResultT} items that are marked as deleted.
      */
     CompletableFuture<List<ResultT>> findAllExisting(long bucketOffset, TimeoutTimer timer) {
-        val result = new HashMap<HashedArray, ResultT>();
+        val result = new HashMap<BufferView, ResultT>();
```

Comment:
`HashMap` only invokes `hashCode` once and then remembers that value for as long as the (key) object is inside it. So whether it's cached or not, it makes little difference as it will only be used once.

Reply:
It is recomputed if the hashmap is resized. (Which should be infrequent, but it's worth noting.)

Reply:
OK.